src/common.js

```diff
@@ -56,18 +56,11 @@ async function runKeyguard(RequestApiClass, options) { // eslint-disable-line no
     /** @type {TopLevelApi | IFrameApi} */
     const api = new RequestApiClass();
 
-    /**
-     * @returns {string}
-     */
-    function allowedOrigin() {
-        switch (window.location.origin) {
-        case 'https://keyguard-next.nimiq.com': return 'https://accounts.nimiq.com';
-        case 'https://keyguard-next.nimiq-testnet.com': return 'https://accounts.nimiq-testnet.com';
-        default: return '*';
-        }
-    }
+    /** @type {string} */
+    // eslint-disable-next-line no-undef
+    const allowedOrigin = CONFIG.ALLOWED_ORIGIN ? CONFIG.ALLOWED_ORIGIN : '*';
```

Comment:
When `CONFIG` is not set, this will result in an `Cannot read ALLOWED_ORIGIN of undefined` error.

Better would be 
```
= CONFIG && CONFIG.ALLOWED_ORIGIN ? ...
```

However, since the build script replaces the thing, this won't work. 

I think you should use one variable for the allowed origin, not a nested object.